src/hooks/api.ts

```diff
@@ -0,0 +1,72 @@
+type Method = 'GET' | 'POST' | 'PUT' | 'DELETE';
+
+interface ApiOptions {
+  method?: Method;
+  body?: Record<string, unknown>;
+  headers?: Record<string, string>;
+  token?: string;
+}
+
+interface ApiError extends Error {
+  flash?: Record<string, unknown>;
+  statusCode?: number;
+}
+
+const API_BASE_URL = process.env.NEXT_PUBLIC_BACKEND_URL;
+
+export const apiRequest = async <T>(
+  url: string,
+  { method = 'GET', body, headers = {}, token }: ApiOptions = {}
+): Promise<T> => {
+  const defaultHeaders: Record<string, string> = {
+    'Content-Type': 'application/json',
+  };
+
+  const authHeaders: Record<string, string> = token
+    ? { Authorization: `Bearer ${token}` }
+    : {};
+
+  try {
+    const response = await fetch(`${API_BASE_URL}${url}`, {
+      method,
+      headers: {
+        ...defaultHeaders,
+        ...authHeaders,
+        ...headers,
+      },
+      body: body ? JSON.stringify(body) : undefined,
+    });
```

Comment:
Set `options` directly to the fetch request.

```suggestion
    const response = await fetch(`${API_BASE_URL}${url}`, options);
```